be/src/storage/lake/update_compaction_state.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/lake/update_compaction_state.h"
+
+#include "gutil/strings/substitute.h"
+#include "storage/chunk_helper.h"
+#include "storage/chunk_iterator.h"
+#include "storage/lake/rowset.h"
+#include "storage/primary_key_encoder.h"
+#include "storage/tablet_manager.h"
+
+namespace starrocks::lake {
+
+CompactionState::CompactionState(Rowset* rowset) {
+    if (rowset->num_segments() > 0) {
+        pk_cols.resize(rowset->num_segments());
+    }
+}
+
+CompactionState::~CompactionState() {}
+
+Status CompactionState::load_segments(Rowset* rowset, const TabletSchema& tablet_schema, uint32_t segment_id) {
+    if (segment_id >= pk_cols.size() && pk_cols.size() != 0) {
+        std::string msg = strings::Substitute("Error segment id: $0 vs $1", segment_id, pk_cols.size());
+        LOG(WARNING) << msg;
```

Comment:
Question: why not just `LOG(WARNING) << "Error segment id: " << segment_id << " vs " << pk_cols.size();`